samcli/commands/local/cli_common/invoke_context.py

```diff
@@ -220,6 +226,11 @@ def __enter__(self) -> "InvokeContext":
 
         self._stacks = self._get_stacks()
 
+        if self._ctx:
+            _, matches_output = get_lint_matches(self._template_file, self._ctx.debug, self._aws_region)
```

Comment:
Should we wrap this call with `try/except` since there might be some exceptions which is thrown by linter logic? (I saw we are checking valid region configuration for instance).